kano_qt_apps/lib/src/q_app_list.cpp

```diff
@@ -59,15 +59,25 @@ void QAppList::clean_q_app_list()
 }
 
 
-// TODO: Update the array as it changes to avoid calculating this multiple times
 QList<QVariant> QAppList::get_app_list()
 {
+    return get_app_list(true);
+}
+
+
+// TODO: Update the array as it changes to avoid calculating this multiple times
+QList<QVariant> QAppList::get_app_list(bool filter=true)
+{
     this->clean_q_app_list();
 
     for (auto app : this->app_list) {
         auto q_app = new QApp(app);
-        auto variant_q_app = QVariant::fromValue(q_app);
-        this->q_app_list.push_back(variant_q_app);
+
+        // If filter is requested, add the app unless it is set hidden
+        if ((filter && !q_app->get_hidden()) || !filter) {
```

Comment:
This seems awfully complicated, isn't this the same as

```c++
if (!(filter && q_app->get_hidden())) {
    ...
}
```

or maybe even clearer

```c++
for (...) {
    if (filter && q_app->get_hidden())) {
        continue;
    }

    ...
}
...
```